### DISCUSSION TITLE
How would I implement server-side redirects for the front-end?

### DISCUSSION THREAD

**Discussion by canrozanes** [karlkeefer/pngr#41] [Q&A]
Let's say I'm working with the example CRUD app and want to redirect the root path `/` to `/posts` on the server side. I also want to have all the other front-end routes to work as is. So I still need the root path to continue serving the react-app.

How would I go about implementing that?

I'm not too familiar with nginx but I'm guessing the answer lies within the nginx configurations. 

As a follow-up, looking at the code I can see that there is a [nginx/nginx.conf](https://github.com/karlkeefer/pngr/blob/master/nginx/nginx.conf) as well as [nginx.prod.conf](https://github.com/karlkeefer/pngr/blob/master/nginx/nginx.prod.conf). Would I have to make the relevant changes in both places?

**Comment by karlkeefer** [comment]
Your ask here sounds like you want the root to do two different things. Redirect to /posts and serve the front-end app.

I can't understand why you wouldn't just do a client-side redirect with react-router's Redirect component?

In general, with nginx you can route a path differently using another location directive for that path: https://www.keycdn.com/support/nginx-location-directive

The first of the nignx configs you linked only runs locally, so if whatever is serving /posts also runs on locally them you could add a location block to both files. The redirect target inside of the location directive might be different if e.g. your local app runs on a different port locally, or something like that.

**Comment by canrozanes** [comment]
Thank you for your answer! 

> sounds like you want the root to do two different things. Redirect to /posts and serve the front-end app.

Yes, this is the weird/funny part.

Doing a client-side redirect certainly works. 

The reason i'm asking is, I was reading through react-router [update guide](https://reactrouter.com/en/main/upgrading/v5#upgrade-all-switch-elements-to-routes) from v5 to v6. In it, there is a link to an [article](https://gist.github.com/mjackson/b5748add2795ce7448a366ae8f8ae3bb) that says:

> Our recommendation for redirecting in React Router v6 really doesn't have much to do with React or React Router at all. It is simply this: if you need to redirect, do it on the server before you render any React and send the right status code. That's it.

So it just became a hypothetical question on how to implement this advice on PNGR.

**Comment by karlkeefer** [comment]
Interesting. Well, the extra location directive in nginx should do the trick, then!
